connectors/es/document.py

```diff
@@ -46,3 +49,75 @@ async def reload(self):
         self._seq_no = doc_source.get("_seq_no")
         self._primary_term = doc_source.get("_primary_term")
         self._source = doc_source.get("_source", {})
+
+    def log_debug(self, msg, *args, **kwargs):
```

Comment:
You can probably reduce the code here by implementing a single method in a wrapper class

```
class DocumentLogger:
    def __init__(self, logger):
        self._logger = logger
        setattr(self, f'{level}', functools.partial(self._log,level))   

    def _log(self, level, msg, *args, **kwargs):
       func = getattr(logger, level.lower())
       func(
            msg,
            *args,
            prefix=self._prefix(),  
            extra=self._extra(),
            **kwargs,
        )
```

and then in you code:

```
self.logger = DocumentLogger(logger)
```

so you can use:

```
self.logger.debug('message')
```

Reply:
I actually had similar implementation:
At [the end of __init__](https://github.com/elastic/connectors-python/blob/86c030afb61ef51c4f1447c1c9af9dfd62285553/connectors/es/document.py#L32) of `ESDocument`, add:

```python
for level in [
    "debug",
    "info",
    "warning",
    "error",
    "exception",
    "critical",
    "fatal",
]:
    setattr(self, f"log_{level}", functools.partial(self._log, level))
```

and add the same `_log` method as you proposed. But the problem is that pyright will complain when I call something like `connector.log_debug(...`,:
```shell
/Users/chenhui.wang/src/github.com/elastic/connectors-python/connectors/byoc.py:509:18 - error: Cannot access member "log_debug" for type "Connector"
```

I have to add `pyright: ignore` every time I call `log_*`. I'd rather explicitly define the method for each job level.